core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedClusterMessage.scala

```diff
@@ -30,6 +30,8 @@ private[spark] object CoarseGrainedClusterMessages {
 
   case object RetrieveSparkProps extends CoarseGrainedClusterMessage
 
+  case object RetrieveCurrentExecutorIdCounter extends CoarseGrainedClusterMessage
```

Comment:
I would call this retrieve max executor ID. Without context the reader has no idea what `executorIdCounter` is.